### PR TITLE
Prevent delete operation if published

### PR DESCRIPTION
**Pull request by reggie7**
[577](https://github.com/headwirecom/peregrine-cms/issues/577) - I'm leaving the action button in order to be transparent for the user - this way they will know what is going on:

delete

Please note that I've set java to `11` in the main `pom.xml`. Let me knoe what you think.

### CODE REVIEW

**Comment by GastonGonzalez** [comment]
@reggie7 - I really like what I am seeing. I'm still testing and noticed one edge case so far that does not work. I believe it is a reference issue and technically not an issue with this PR.

Use Case
1. Create a page (p1) and publish it.
2. Create a second page (p2). Add a card component and create a card that links to (p1). Publish p2.
3. Unpublish p1.
4. Delete p1.

Observed Behavior
1. p1 is allowed to be deleted.

Expected Behavior
1. p1 is referenced in p2. It should not be allowed for deletion.

**Comment by GastonGonzalez** [comment]
@reggie7 - This PR is very difficult to determine the exact changes due to the number of clean up updates. Can you please remove the clean up updates. We need to avoid refactoring in these types of tickets. Refactoring should be done in a separate ticket. Also, please do not use wildcard imports.

**Comment by GastonGonzalez** [comment]
@reggie7 - This change has a performance impact. https://github.com/headwirecom/peregrine-cms/pull/585/files#diff-2ca240fef4c0c7351fe4a53bd9a3ee13R376. This looks like another refactoring. It should be reverted.

**Comment by GastonGonzalez** [comment]
@reggie7 - What would the effort be to change the toast popup when the delete button is clicked on a published resource to prompt a modal with the same message as well as show the references?

**Comment by GastonGonzalez** [comment]
@reggie7 - The page move is also a problem. I think this can be a separate ticket.

Use Case
1. Create page (p1) and publish
2. Move p1 to a new location

Observed Behavior
1. p1 is moved but published content is not left on cache or publish/live server

Expected Behavior
1. We need to message the user that this page is published and needs to be unpublished to support the move. A broader conversion on Slack is likely needed.

**Comment by GastonGonzalez** [comment]
> @reggie7 - What would the effort be to change the toast popup when the delete button is clicked on a published resource to prompt a modal with the same message as well as show the references?

@reggie7 - Let's make this PR simple. Let's just keep the toast message/functionality as-is.

**Comment by GastonGonzalez** [comment]
> @reggie7 - The page move is also a problem. I think this can be a separate ticket.
> 
> Use Case
> 
> 1. Create page (p1) and publish
> 2. Move p1 to a new location
> 
> Observed Behavior
> 
> 1. p1 is moved but published content is not left on cache or publish/live server
> 
> Expected Behavior
> 
> 1. We need to message the user that this page is published and needs to be unpublished to support the move. A broader conversion on Slack is likely needed.

@reggie7 - We will create another ticket for this. We'll keep it simple in the new ticket. Per @reusr1 , we will just prevent any behaviors from being performed that can lead to an unstable state (i.e. move and rename).

**Comment by reggie7** [comment]
> I really like what I am seeing. I'm still testing and noticed one edge case so far that does not work. I believe it is a reference issue and technically not an issue with this PR.

Hmm, true, haven't considered this scenario here. I say we make a separate ticket for this too, what do you think? Or want me to rather include it here already?

**Comment by reggie7** [comment]
All other comments are taken care of (or will be in separate tickets). Looks much cleaner now indeed. Just please use diff with whitespace comparison off - I see there's a file with line endings changed from Windows to Unix style, let's keep that in.

**Comment by GastonGonzalez** [comment]
> > I really like what I am seeing. I'm still testing and noticed one edge case so far that does not work. I believe it is a reference issue and technically not an issue with this PR.
> 
> Hmm, true, haven't considered this scenario here. I say we make a separate ticket for this too, what do you think? Or want me to rather include it here already?

@reggie7 - Please create a separate ticket. I would like to limit the number of changes in this PR.

File: admin-base/core/src/main/java/com/peregrine/admin/servlets/DeletePageServlet.java [hunk 1]
```diff
@@ -75,31 +78,41 @@
 
     public static final String FAILED_TO_DELETE_NODE = "Failed to delete node: ";
 
-    @Reference
-    ModelFactory modelFactory;
```
Comment (GastonGonzalez) [inline admin-base/core/src/main/java/com/peregrine/admin/servlets/DeletePageServlet.java:79]: @reggie7  - Was the ModelFactory not used in this class?

**Comment by GastonGonzalez** [comment]
@reggie7 - I retested your new branch: `issues/577-prevent-delete-operation-if-published-simplified`. It looks good. Please create a new PR off the new branch. I will close this PR without merging it.

**Review by GastonGonzalez — APPROVED** [review]

**Pull request merged**